spec/use_cases/reservations/cancel_spec.rb

```diff
@@ -1,19 +1,29 @@
 require 'rails_helper'
 
 RSpec.describe Reservations::UseCases::Cancel do
-  describe ".call" do
+  describe '.call' do
+    let(:seats) { [*'10'..'13'] }
     let(:movie) { create(:movie) }
     let(:hall) { create(:hall) }
+    let(:user) { create(:user) }
     let(:screening) { create(:screening, hall:, movie:, start_time: Date.current + 2.days) }
-    let(:reservation) { create(:reservation, screening:, status: :booked) }
-    let(:params) do 
-      { id: reservation.id }
+    let(:reservation) do
+      Reservations::UseCases::Create.new(screening_id: screening.id,
+                                         seats:,
+                                         user_id: user.id,
+                                         status: :booked).call
     end
 
-  subject(:cancel_reservation) { described_class.new(**params).call }
+    let(:params) do
+      {
+        id: reservation.id
+      }
+    end
+
+    let(:cancel_reservation) { described_class.new(**params).call }
```

Comment:
you don't need `**` operator :) 
```suggestion
    let(:cancel_reservation) { described_class.new(params).call }
```

Reply:
Actually it does not work without that double splat operator. i get that error `ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: id)`

Reply:
ok because the initializer looks like this:
```ruby
def initialize(id:, repository: ReservationRepository.new)
```

So you should pass arguments like this:
```ruby
let(:cancel_reservation) { described_class.new(id: id).call }
```
Another thing is that your usecase doesn't care about params so you can just define id variable
```ruby
let(:id) { reservation.id }
```